webui/react/src/pages/SignIn.tsx

```diff
@@ -0,0 +1,80 @@
+import { notification } from 'antd';
+import queryString from 'query-string';
+import React, { useEffect } from 'react';
+
+import AuthToken from 'components/AuthToken';
+import DeterminedAuth from 'components/DeterminedAuth';
+import Logo, { LogoTypes } from 'components/Logo';
+import Auth from 'contexts/Auth';
+import FullPageSpinner from 'contexts/FullPageSpinner';
+import useAuthCheck from 'hooks/useAuthCheck';
+import usePolling from 'hooks/usePolling';
+import { routeAll } from 'routes';
+
+import css from './SignIn.module.scss';
+
+interface Queries {
+  cli?: boolean;
+  redirect?: string;
+}
+
+const DEFAULT_REDIRECT = '/det/dashboard';
+
+const SignIn: React.FC = () => {
+  const auth = Auth.useStateContext();
+  const showSpinner = FullPageSpinner.useStateContext();
+  const setShowSpinner = FullPageSpinner.useActionContext();
+  const queries: Queries = queryString.parse(location.search);
+
+  /*
+   * Check every so often to see if the user is authenticated.
+   * For example, the user can authenticate in a different session,
+   * and this will pick up that auth and automatically redirect them into
+   * their previous app.
+   */
+  // const task = useAsyncTask(checkAuth);
+  const checkAuth = useAuthCheck();
+  const stopPolling = usePolling(checkAuth);
```

Comment:
tried the cookie check and it created a redirect loop because the `Route` component no longer preforms the auth verification. It only detects that an auth is needed but the auth state is not authenticated.